numba/cuda/compiler.py

```diff
@@ -37,7 +37,7 @@ def compile_cuda(pyfunc, return_type, args, debug, inline):
     flags.set('no_compile')
     flags.set('no_cpython_wrapper')
     if debug:
-        flags.set('boundcheck')
+        flags.set('boundscheck')
```

Comment:
We should let this be off and maybe check that the flag is never turned on.

Reply:
It already raises an error if the flag is set (there is a test for this).